simple_salesforce/api.py

```diff
@@ -493,43 +493,61 @@ def __init__(
                                      object_name=object_name,
                                      sf_version=sf_version))
 
-    def metadata(self):
+    def metadata(self, headers=None):
         """Returns the result of a GET to `.../{object_name}/` as a dict
         decoded from the JSON payload returned by Salesforce.
+
+        Arguments:
+
+        * headers -- a dict with additional request headers.
         """
-        result = self._call_salesforce('GET', self.base_url)
+        result = self._call_salesforce('GET', self.base_url, headers=headers)
         return result.json(object_pairs_hook=OrderedDict)
 
-    def describe(self):
+    def describe(self, headers=None):
         """Returns the result of a GET to `.../{object_name}/describe` as a
         dict decoded from the JSON payload returned by Salesforce.
+
+        Arguments:
+
+        * headers -- a dict with additional request headers.
         """
-        result = self._call_salesforce('GET', self.base_url + 'describe')
+        result = self._call_salesforce(
+            'GET', self.base_url + 'describe', headers=headers)
```

Comment:
Nitpicky, but I'd prefer to see either interpolation or `urljoin` used here instead of addition (even though it's used elsewhere): `'%s/%s' % (self.base_url, 'describe',)`